lib/src/string_compare.dart

```diff
@@ -20,13 +20,32 @@ bool _isWhitespace(int c) {
       c == 0xfeff; // ZERO WIDTH NO_BREAK SPACE.
 }
 
-/// Returns the index of the next non-whitespace character.
+/// Returns the next non-whitespace character.
 ///
-/// Returns `true` if current contains a non-whitespace character.
-/// Returns `false` if no characters are left.
+/// Searches [str] for non-whitespace characters, starting at [i]
+/// and ending at [len].
+///
+/// Skips whitespace characters *and* single backslashes before whitespace
+/// characters. The latter is allowed to account for the first line
+/// of multiline strings (but does risk allowing the removal of a backslash
+/// from a raw string.)
```

Comment:
Discarding whitespace on the first line seems reasonable to me. We discard trailing whitespace in other places where it isn't semantic.

But treating an escaped space as a whitespace character and removing that feels like a step a little too far. We don't do escape normalization anywhere else (there is an argument that we should), so I think it's probably best to leave those alone.

In practice, I can't imagine any reasonable human writing an escaped space on the first line of a multi-line string anyway, so it's probably fine to not support cleaning that up.

Reply:
Ended up normalizing everything around the backslashes, but retaining them (for absolutely no effect, they are effectively whitespace, that is ignored by everything that comes after).